update logs to save & connectAndDiscover

mark-prins opened the pull request:
Fixes #4045 #4055 

## Change summary

Disconnect and reconnect code from the cold chain app, which seems to resolve the problem with a ` BleError: Device MA:CA:DD:RE:SS was disconnected` error being thrown quietly, which was preventing logs from being downloaed.

Have included the fix ( from @joshxg :) )for #4045 too.

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Download some logs; turn off the tablet for a while. turn on and hopefully, get some more logs.

### Related areas to think about

If there are any general areas of the codebase your changes might have side affects on, mention them here

katherine-sussol reviewed (dismissed):
Ohh is that what that code was for in the cce app!
Feeding two 🐦  with one scone, nice one.

joshxg reviewed (dismissed):
Cool! 

Would it be better to put this into master so we can do a quick patch tomorrow?

katherine-sussol commented on src/bluetooth/BleService.js:
```diff
@@ -52,11 +52,16 @@ class BleService {
     }
 
-    return this.manager.discoverAllServicesAndCharacteristicsForDevice(macAddress);
+    const device = await this.connectToDevice(macAddress, { autoConnect: true });
```
Hm, does the `autoConnect` arg get passed through to the BleManager if `connectToDevice` doesn't also pass  through? 🤔  https://github.com/openmsupply/mobile/blob/1f2149f1c6861883579072786747316b4fc9915e/src/bluetooth/BleService.js#L44

mark-prins replied:
DOH! good spotting @katherine-sussol

mark-prins commented:
> Would it be better to put this into master so we can do a quick patch tomorrow?

Yes, probably. Was still rather hopeful that we could get the select-a-vaccinator-and-crash-the-app feature removed and do a full release, though this fix is probably good to release sooner rather than later. Would be nice to test it with them.

mark-prins commented:
@joshxg @katherine-sussol oops! I changed to master to see how bad that would be.. and it wasn't good, so I've switched back. but that had dismissed the reviews.. sorry. can you click again please? I'll make another PR for master 🙄

mark-prins commented:
actually.. I might have another look first:

![gap-in-graph](https://user-images.githubusercontent.com/9192912/118889655-85032f00-b951-11eb-9a85-e7a19684532c.png)

joshxg commented:
oh noooo...

But that looks like it did download some logs this morning ?

mark-prins commented:
With further testing I can't repro - seems worth getting this out to the client and testing with them.

mark-prins commented:
@katherine-sussol or @joshxg would you mind re-reviewing? I'd like to merge this into develop before I forget

joshxg commented:
@mark-prins I've just merged master into develop which I think makes this redundant now?

mark-prins commented:
Quite right, thanks
